multiqc/utils/report.py

```diff
@@ -455,30 +490,29 @@ def exclude_file(sp, f):
     # Search by file name (glob)
     if "exclude_fn" in sp:
         for pat in sp["exclude_fn"]:
-            if fnmatch.fnmatch(f["fn"], pat):
+            if fnmatch.fnmatch(f.filename, pat):
                 return True
 
     # Search by file name (regex)
     if "exclude_fn_re" in sp:
         for pat in sp["exclude_fn_re"]:
-            if re.match(pat, f["fn"]):
+            if re.match(pat, f.filename):
                 return True
 
     # Search the contents of the file
     if "exclude_contents" in sp or "exclude_contents_re" in sp:
         # Compile regex patterns if we have any
         if "exclude_contents_re" in sp:
             sp["exclude_contents_re"] = [re.compile(pat) for pat in sp["exclude_contents_re"]]
-        with io.open(os.path.join(f["root"], f["fn"]), "r", encoding="utf-8") as fh:
-            for line in fh:
-                if "exclude_contents" in sp:
-                    for pat in sp["exclude_contents"]:
-                        if pat in line:
-                            return True
-                if "exclude_contents_re" in sp:
-                    for pat in sp["exclude_contents_re"]:
-                        if re.search(pat, line):
-                            return True
+        for line in f.line_iterator():
```

Comment:
It appears to be a second call of `line_iterator` within the same `with` call, so it would probably read from the place where `search_file` has left it?

Reply:
Right, I mixed two distinct context managers here - the FileSize object that wraps the file handler, and the line_iterator that wraps content_lines 👍

Reply:
Added some docstring comments to clarify that, mainly for myself in the future 😅